### ISSUE TITLE
There is no output on the node

### ISSUE THREAD

**Issue by gdiciancia** [Bouni/node-red-contrib-luxtronik2-ws#2]
Hi @Bouni,
The Node status shows its going through all sections (i.e Temperatures), but there's no output from the node.

I saw this error, which is crashing/restarting node-red:

11 Aug 13:18:08 - [red] Uncaught Exception:
11 Aug 13:18:08 - TypeError: Cannot set property '[object Array]' of undefined
    at /opt/node_modules/node-red-contrib-luxtronik2-ws/luxtronik2-ws.js:80:50

Thank you for your help

**Comment by Bouni** [comment]
I assume that the structure of the output changed and therefor the script fails.
As I no longer use node-red, I don't have a setup to debug.
If you add a debug line before this line https://github.com/Bouni/node-red-contrib-luxtronik2-ws/blob/master/luxtronik2-ws.js#L48 you shoud see the data that is returned.
Post that data here and I can guide you.

**Comment by rlindner** [comment]
@gdiciancia As you've asked in my pull request: I'm using an timestamp inject node with an interval of 5 seconds and a debug output node which logs `msg.payload`. The version of my Luxtronik is V2.81 and it's working pretty well.

**Comment by Bouni** [comment]
The PR of @rlindner didn't changed any logic so it seems that the problem comes from somewhere else. What version of Luxtronik do you have running @gdiciancia ?

**Comment by gdiciancia** [comment]
Hi both, first thanks for the replies.  @Bouni, I have version 3.82.2.
@rlindner, I have the same, but no output. Seems to be a mapping problem.
I will try to debug (assuming I can find the directory where your node files are stored and can change them directly). 
FYI. I'm running node-red as an add-on to home assistant, which runs on docker and I have no experience yet developing/debugging node-red nodes in particular.

**Comment by gdiciancia** [comment]
I was trying in parallel to get the output with a python script (https://pastebin.com/0ruYFjmW), but somehow it is not connecting.

**Comment by gdiciancia** [comment]
This is the Temperatures response:
`<?xml version="1.0"?><Content id='0x26eca0'><name>Settings</name><item id='0x2a1260'><name>Mode of operation</name><item id='0x2363e8'><name>Heating circuit</name><option value='0'>Auto</option><option value='1'>Second heat gen.</option><option value='2'>Party</option><option value='3'>Holidays</option><option value='4'>Off</option><raw>4</raw><value>Off</value></item><item id='0x235de0'><name>service water</name><option value='0'>Auto</option><option value='1'>Second heat gen.</option><option value='2'>Party</option><option value='3'>Holidays</option><option value='4'>Off</option><raw>0</raw><value>Auto</value></item><item id='0x2a1208'><name>Cooling</name><option value='0'>Off</option><option value='1'>Auto</option><raw>1</raw><value>Auto</value></item></item><item id='0x2a11b0'><name>Temperatures</name><item id='0x2360f8'><name>Limit return</name><value>56.0°C</value></item><item id='0x2a0ff8'><name>Hysteres HC</name><value>3.0 K</value></item><item id='0x27dd98'><name>hysteres.SW</name><value>10.0 K</value></item><item id='0x235fa0'><name>RTincr.e max</name><value>3.0 K</value></item><item id='0x26e260'><name>release 2hg</name><value>-16.0°C</value></item><item id='0x26ed78'><name>TVth.disinf2</name><min>500</min><max>800</max><step>10</step><unit>°C</unit><div>10.00</div><raw>600</raw><value>60.0°C</value></item><item id='0x26f7f8'><name>T-HS min</name><value>-9.0°C</value></item><item id='0x237f68'><name>T-HG max</name><value>120.0°C</value></item><item id='0x2acda0'><name>lowering to</name><min>-200</min><max>100</max><step>5</step><unit>°C</unit><div>10.00</div><raw>-200</raw><value>-20.0°C</value></item><item id='0x2a12b8'><name>flow max.</name><value>62.0°C</value></item><item id='0x235e38'><name>min HSI flow max</name><value>0.0°C</value></item><item id='0x2a1490'><name>flow limit of applic.</name><value>56.0°C</value></item><item id='0x2478e0'><name>service water max </name><value>55.0°C</value></item><item id='0x23f768'><name>min.systemtemp.</name><min>150</min><max>300</max><step>5</step><unit>°C</unit><div>10.00</div><raw>150</raw><value>15.0°C</value></item><item id='0x23f7b0'><name>min. flow cooling</name><value>18.0°C</value></item></item><item id='0x2a1438'><name>System settings</name><item id='0x27db70'><name>El. sup. blockade</name><value>no 2hg</value></item><item id='0x27dbc8'><name>room stat.</name><option value='0'>No</option><option value='4'>RBE</option><option value='6'>Smart</option><raw>0</raw><value>No</value></item><item id='0x234ea0'><name>Integration</name><value>Return</value></item><item id='0x234ee8'><name>mixing circ 1</name><option value='0'>No</option><option value='1'>Discharge</option><option value='2'>Charge</option><option value='3'>Cool</option><option value='4'>HW+Cool</option><raw>3</raw><value>Cool</value></item><item id='0x26edf0'><name>Error</name><value>with 2hg</value></item><item id='0x2382b8'><name>service water 1</name><option value='0'>Sensor</option><option value='1'>Therm.</option><raw>0</raw><value>Sensor</value></item><item id='0x26ef30'><name>service water 2</name><value>CP</value></item><item id='0x26ef78'><name>service water 3</name><value>with CP</value></item><item id='0x234ce0'><name>service water 4</name><value>set value</value></item><item id='0x234d78'><name>service water 5</name><value>with HSP</value></item><item id='0x264180'><name>SW+HP max</name><min>0</min><max>80</max><step>5</step><unit> h</unit><div>10.00</div><raw>0</raw><value>0.0 h</value></item><item id='0x236c30'><name>pump opt.</name><option value='0'>No</option><option value='1'>Yes</option><raw>0</raw><value>No</value></item><item id='0x234a30'><name>Access</name><value>AS</value></item><item id='0x234a78'><name>brine pres/flow</name><value>pow.suppl.</value></item><item id='0x234b08'><name>control CP</name><value>Off</value></item><item id='0x247590'><name>setting hc</name><option value='0'>set. AT</option><option value='1'>fixed Tp</option><raw>0</raw><value>set. AT</value></item><item id='0x2475d8'><name>setting mc 1</name><option value='0'>set. AT</option><option value='1'>fixed Tp</option><raw>0</raw><value>set. AT</value></item><item id='0x26fb20'><name>screed heating</name><option value='0'>w. mixer</option><option value='1'>wo. Mixer</option><raw>0</raw><value>w. mixer</value></item><item id='0x26fb68'><name>El. Anode</name><value>No</value></item><item id='0x264730'><name>heating limit</name><option value='0'>No</option><option value='1'>Yes</option><raw>1</raw><value>Yes</value></item><item id='0x2647c8'><name>parall. operation</name><value>No</value></item><item id='0x26f180'><name>remote maintenance</name><option value='0'>No</option><option value='1'>Yes</option><raw>0</raw><value>No</value></item><item id='0x26f1c8'><name>pump optim. Time</name><min>5</min><max>180</max><step>5</step><unit> min</unit><div>1.00</div><raw>180</raw><value>180 min</value></item><item id='0x2352a8'><name>TDI Message</name><value>No</value></item><item id='0x244d50'><name>heat source</name><value>Brine</value></item><item id='0x244de0'><name>release 2hg</name><value>60 min</value></item><item id='0x244e28'><name>aux. heat. hot water</name><option value='0'>No</option><option value='1'>Yes</option><raw>0</raw><value>No</value></item><item id='0x244e70'><name>aux.serv.water max</name><raw>8</raw><value>5.0 h</value></item><item id='0x244f90'><name>smart grid</name><value>No</value></item><item id='0x245058'><name>setting mc 1</name><option value='0'>fast</option><option value='1'>medium</option><option value='2'>slow</option><raw>0</raw><value>fast</value></item></item></Content>`

**Comment by Bouni** [comment]
> FYI. I'm running node-red as an add-on to home assistant, which runs on docker and I have no experience yet developing/debugging node-red nodes in particular.

😵  Why don't you simply use my [custom Home Assistant component](https://github.com/Bouni/Home-Assistant-Luxtronik) that integrates Luxtronik directly?

Leave the pain in the A**** of using node-red (personal opinion) behind 😁

**Comment by gdiciancia** [comment]
I hear you :), the reason I like the new method with sockets is that the data provides all the attributes based on the language selected and there's no mapping file required.
I changed the following locally and then it worked (the returned value for the "Access: User" menu created the error):

`                     if (json.Navigation.item[i].name[0] == 'Access: User') { //new
                        node.status({fill:"green",shape:"dot",text:"Skipping: "+json.Navigation.item[i].name[0]});
                    } else { //new
                        var name = json.Navigation.item[i].name[0];
                        var id = json.Navigation.item[i].$.id;
                        msg.payload[name] = {};
                        node.ws.send('GET;'+id);
                        node.count++;
                    } //new
`

**Comment by Bouni** [comment]
@gdiciancia If thats better for youm sure go for it. 
I for myself never regreted that I droped nodejs and it ecosystem in favor of Home Assistant and Python 😎

**Comment by gdiciancia** [comment]
@Bouni, the good thing of the component is it can create the entities in HA, would be nice to have WS version of the HA component ;)
Ok, I'll try to create the PR.
Thanks again for all the feedback.

**Comment by Bouni** [comment]
>would be nice to have WS version of the HA component

I thought about that in the first place but then I realized that Alpha Innotec didn't disabled the binary protocoll on newer heatpumps as I thought. So by using the binary protocoll I can support all pumps, not just newer ones. And they use the binary protocoll in their Android App so its quite safe to assume it will not be dumped anywhere soon.

**Comment by gdiciancia** [comment]
PR created.
